src/write.rs

```diff
@@ -330,6 +357,52 @@ where
     pub fn raw_write_handle(&mut self) -> NonNull<T> {
         self.w_handle
     }
+
+    /// Returns the backing data structure.
+    ///
+    /// Makes sure that all the pending operations are applied and waits till all the read handles
+    /// have departed. Then it uses `Absorb::drop_first` to drop one of the copies of the data and
+    /// returns the other copy as an owned value.
+    ///
+    /// ## With custom `drop_second`:
```

Comment:
I'm very glad you wrote this section — was about to ask about it. I wonder if we want to provide a wrapper type ourselves? Something like:
```rust
struct Absorbing<T: Absorb>(Box<T>);
// impl Deref + DerefMut for Absorbing
impl<T: Absorb> Absorbing<T> {
    /// This is unsafe because you must call Absorb::drop_second
    pub unsafe fn into_box(self) -> Box<T> {
    }
}
impl<T: Absorb> Drop for Absorbing<T> {
    fn drop(&mut self) {
        T::drop_second(self);
    }
}
```
What do you think? The name is maybe a little weird — happy to hear alternatives. I think providing this type makes it much clearer to the caller exactly what they need to do, and tries to enforce that through the type system.

Reply:
It seems like a much safer version. And this also makes the drop implementation a little nicer.

For the naming:
* it only adds extra functionality to `drop_second`
* this is taken owned data

I think good options are: `AbsorbDrop` and `Taken`
I am going to implement it with `AbsorbDrop` for now and we can change it if something better is found.